Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -1470,9 +1469,25 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
 
     // MARK: - Connection Tester
 
-    private enum ConnectionTesterError: Error {
+    private enum ConnectionTesterError: CustomNSError {
         case couldNotRetrieveInterfaceNameFromAdapter
         case testerFailedToStart(internalError: Error)
+
+        var errorCode: Int {
+            switch self {
+            case .couldNotRetrieveInterfaceNameFromAdapter: return 0
+            case .testerFailedToStart: return 1
+            }
+        }
+
+        var errorUserInfo: [String: Any] {
+            switch self {
+            case .couldNotRetrieveInterfaceNameFromAdapter:
+                return [:]
+            case .testerFailedToStart(let internalError):
+                return [NSUnderlyingErrorKey: internalError as NSError]
+            }
+        }
```

Comment:
We were missing wake error information from our pixels since this didn't implement `CustomNSError` so I just added that to avoid another review.